test/tc_basic.rb

```diff
@@ -114,7 +114,7 @@ def test_set_and_get_point_from_wkt
               obj_ = klass_.new
               assert_nil(obj_.latlon)
               obj_.latlon = 'SRID=1000;POINT(1 2)'
-              assert_equal(@factory.point(1, 2), obj_.latlon)
+              assert_equal(RGeo::Cartesian.preferred_factory(srid: 1000).point(1, 2), obj_.latlon)
```

Comment:
@dschweisguth yes I had geos installed, version 3.5. After uninstalling geos those tests fail:

```bash
  1) Failure:
RGeo::ActiveRecord::Mysql2SpatialAdapter::Tests::TestBasic#test_readme_example [test/tc_basic.rb:165]:
Expected: true
  Actual: false

  2) Failure:
RGeo::ActiveRecord::Mysql2SpatialAdapter::Tests::TestBasic#test_save_and_load_point_from_wkt [test/tc_basic.rb:141]:
No visible difference in the RGeo::Cartesian::PointImpl#inspect output.
You should look at the implementation of #== on RGeo::Cartesian::PointImpl or its members.
#<RGeo::Cartesian::PointImpl:0xXXXXXX "POINT (1.0 2.0)">

  3) Failure:
RGeo::ActiveRecord::Mysql2SpatialAdapter::Tests::TestBasic#test_set_and_get_point_from_wkt [test/tc_basic.rb:117]:
No visible difference in the RGeo::Cartesian::PointImpl#inspect output.
You should look at the implementation of #== on RGeo::Cartesian::PointImpl or its members.
#<RGeo::Cartesian::PointImpl:0xXXXXXX "POINT (1.0 2.0)">
```

I installed it again (this time version 3.6) and tests pass again.

Now, installation of `geos` is 'highly' recommended. For older versions of rgeo gem it wasn't even possible to install it without geos present.

To sum up, I see two distinct topics in this PR:
1. gemspec, Gemfile and gitignore cleanup, which could be done in just 2 commits (1 for gems, 1 for gitignore)
2. dealing with rgeo and geos - honestly, I'd skip that and just focus on support for ActiveRecord 4.2+

You already spent some time on this and I think that it'd be nice if your commits make their way to master. Could you please remove commits related to rgeo from this branch and squash it to 2 commits. Then we could release 0.5.1 version of the gem.